agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -95,6 +132,8 @@ def ensure_desired_state
       if service_pod.running? && service_container.nil?
         info "creating #{service_pod.name}"
         ensure_running
+      elsif service_pod.running? && (service_container && service_container.running? && health.unhealthy?)
+        ensure_healing
```

Comment:
Shouldn't the `service_container_outdated?` check come before the `unhealthy?` check? This would block deployments for an unhealthy service?